test/intl402/DisplayNames/prototype/of/type-language-invalid.js

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2023 Igalia S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-Intl.DisplayNames.prototype.of
+description: Throws a RangeError for invalid `language` codes
+info: |
+  12.3.3 Intl.DisplayNames.prototype.of ( code )
+
+  1. If type is "language", then
+      a. If code cannot be matched by the unicode_language_id Unicode locale nonterminal, throw a RangeError exception.
+      b. If IsStructurallyValidLanguageTag(code) is false, throw a RangeError exception.
+      c. Return CanonicalizeUnicodeLocaleId(code).
+features: [Intl.DisplayNames-v2]
+---*/
+
+var displayNames = new Intl.DisplayNames(undefined, {type: 'language'});
+
+assert.throws(RangeError, function() {
+  displayNames.of('');
+}, 'invalid language subtag - empty string');
+
+assert.throws(RangeError, function() {
+  displayNames.of('a');
+}, 'invalid language subtag - only one character');
+
+assert.throws(RangeError, function() {
+  displayNames.of('abcdefghi');
+}, 'invalid language subtag - greater than 8 characters');
+
+assert.throws(RangeError, function() {
+  displayNames.of('en-u-hebrew');
+}, 'invalid language id -- contains singleton');
+
+assert.throws(RangeError, function() {
+  displayNames.of('aa-aaaa-bbbb');
+}, 'invalid language id -- contains multiple script subtags');
+
+assert.throws(RangeError, function() {
+  displayNames.of('aa-aaaaa-aaaaa');
+}, 'duplicate variant subtag');
+
+assert.throws(RangeError, function() {
+  displayNames.of('aa-aa-aa');
+}, 'more than two subtags of length 2');
+
+assert.throws(RangeError, function() {
+  displayNames.of('00');
+}, 'numerical subtag of length 2');
+
+assert.throws(RangeError, function() {
+  displayNames.of('0000');
+}, 'numerical subtag of length 4');
+
+assert.throws(RangeError, function() {
+  displayNames.of('@#$%@#$');
+}, 'invalid characters');
+
+assert.throws(RangeError, function() {
+  displayNames.of('en-US-');
+}, 'separator not followed by subtag');
+
+assert.throws(RangeError, function() {
+  displayNames.of('-en');
+}, 'separator at start of language id');
+
+assert.throws(RangeError, function() {
+  displayNames.of('en--GB');
+}, 'separators with no subtag between them');
+
+assert.throws(RangeError, function() {
+  displayNames.of('root');
+}, 'tries to use backwards compatibility feature ("root" instead of "und")');
+
+assert.throws(RangeError, function(){
+  displayNames.of('abcd-GB');
+}, 'tries to use backwards compatibility feature (script subtag without a language subtag)');
+
+assert.throws(RangeError, function(){
+  displayNames.of('abcd');
+}, 'tries to use backwards compatibility feature (bare script subtag)');
+
+assert.throws(RangeError, function(){
+  displayNames.of('en_GB');
+}, 'tries to use backwards compatibility feature (_ as separator)');
```

Comment:
"backwards compatibility" seems like a _good_ thing; we should probably make the issues with it more clear in these messages.

```suggestion
assert.throws(RangeError, function() {
  displayNames.of('root');
}, 'BCP 47-incompatible CLDR syntax ("root" instead of "und")');

assert.throws(RangeError, function(){
  displayNames.of('abcd-GB');
}, 'BCP 47-incompatible CLDR syntax (script subtag without a language subtag)');

assert.throws(RangeError, function(){
  displayNames.of('abcd');
}, 'BCP 47-incompatible CLDR syntax (bare script subtag)');

assert.throws(RangeError, function(){
  displayNames.of('en_GB');
}, 'BCP 47-incompatible CLDR syntax (_ as separator)');
```